File: src/27.移除元素.cpp
/*
 * @lc app=leetcode.cn id=27 lang=cpp
 *
 * [27] 移除元素
 */
extern int apppp;
// @lc code=start
#include <iostream>
#include <vector>
class Solution {
public:
    int removeElement(std::vector<int>& nums, int val)
    {
        int slow = 0;
        int fast = 0;
        while (fast < nums.size()) {
            if (nums[fast] != val) {
                nums[slow] = nums[fast];
                slow++;
                fast++;
            } else {
                fast++;
            }
        }
        return slow;
    }
};
// @lc code=end

int main()
{
    Solution s;
    std::vector<int> nums = { 0, 1, 2, 2, 3, 0, 4, 2 };
    int res = s.removeElement(nums, 2);
    std::cout << res << std::endl;
}

File: 数组/27.移除元素.py
from typing import List


class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        slow = 0
        fast = 0
        while fast < len(nums):
            if nums[fast] != val:
                nums[slow] = nums[fast]
                slow += 1
                fast += 1
            else:
                fast += 1
        return slow


if __name__ == "__main__":
    s = Solution()
    res = s.removeElement([0, 1, 2, 2, 3, 0, 4, 2], 2)
    print(res)
